packages/front-end/components/Settings/EditDataSource/ExperimentAssignmentQueries/AddEditExperimentAssignmentQueryModal.tsx

```diff
@@ -29,6 +33,12 @@ export const AddEditExperimentAssignmentQueryModal: FC<EditExperimentAssignmentQ
   onSave,
   onCancel,
 }) => {
+  const [queryError, setQueryError] = useState<null | string>();
```

Comment:
Yea - the error handling is a little weird.

Currently, the way it's built, any errors (missing required columns, unknown columns, etc) are caught and passed into the `Modal` component via the `error` prop.

`DisplayTestQueryResults` doesn't actually display errors, it only displays a success message, and non-fatal warnings (e.g. no rows returned, optional columns passed in, etc).

I originally did this for a better UX as it felt like an anti-pattern to have error messages displayed in multiple spots (since, for example, a missing required column error is handled via the `validateSQL` method and the error is handled by the modal component out of the box.).